volttron/platform/dbutils/sqlitefuncts.py

```diff
@@ -114,6 +115,16 @@ def __init__(self, connect_params, table_names):
 
     def setup_historian_tables(self):
 
+        result = self.select('''PRAGMA auto_vacuum''')
+        auto_vacuum = result[0][0]
+
+        if auto_vacuum != 1:
+            _log.info("auto_vacuum set to 0 (None), updating to 1 (full).")
+            _log.info("VACCUMing DB to cause new auto_vacuum setting to take effect. "
+                      "This could be slow on a large database.")
+            self.select('''PRAGMA auto_vacuum=1''')
+            self.select('''VACUUM;''')
```

Comment:
Wouldn't running vacuum on production database take a long time? And is vacuum necessary even when no history limit or storage limit is set. If they are not set, there aren't going to be any deletes and so not much to vacuum ? Also setting auto_vacuum=1 would save disk space but would slow down writes. Do we need to run the performance test for this change to see if there is not too much of a slow down?

Reply:
It could, but I'm banking on the fact that no one is currently running the SQLite historian in production as right now it's a bad idea.

This is a workaround for not setting auto_vacuum in the first place. We could forgo it if both settings are None but we need to account for those setting changing during run time via the config store. (Yes, the config store is not supported yet, but it will be eventually) Running vacuum after we've started running (and potentially after the DB has become quite large before we decide to change it) would be a bad idea.

It will only happen once ever per database and only at agent startup. From then on out the size will be managed via auto_vacuum. New databases vacuum instantly so there is no worry there.

I've tested the performance of auto_vacuum when I turned it on for the historian cache. The performance difference is almost imperceptible. The size of the database increases by less than one percent. If anything hurts performance it's the loop to manage the size in GB and even that is not that bad as I save the commit for the end.